allofplos/transformations.py

```diff
@@ -94,15 +94,14 @@ def filename_to_doi(filename):
     :param filename: relative path to local XML file in the get_corpus_dir() directory
     :return: full unique identifier for a PLOS article
     """
-    filename = os.path.basename(filename)
-    if correction in filename and validate_filename(filename):
+    fn = os.path.basename(filename)
```

Comment:
couldn't you change the `filename`parameter to `filepath` and then keep this as `filename`?

I always feel conflicted about `fn` since it can also be shorthand for `function`, especially in callback based code.